### ISSUE TITLE
[CI] Frequent failures of MlJobIT

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#45587]
We are seeing frequent incidences of `org.elasticsearch.xpack.ml.integration.MlJobIT` failing. It seems often when it does it's many tests in this suite fail for some root cause. We've had over 500 individual test failures related to this test class in the last 7 days. I'm reluctant to mute the entire suite as it seems like it'd be a significant loss of coverage. Hopefully this is something we can quickly sort out.

https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(branch,test,build_url,build-id),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:kuery,query:'class%20:%20%22org.elasticsearch.xpack.ml.integration.MlJobIT%22'),sort:!(time,desc))

**Comment by davidkyle** [comment]
Thanks for raising this @mark-vieira, it looks like something significant changed around the 9th August causing a jump in failure rate. I'll dig in.

**Comment by droberts195** [comment]
> We've had over 500 individual test failures related to this test class in the last 7 days

All but one of these were between 9th and 12th August.  Most of them are duplicates of #45405, which really related to all tests in the `native-multi-node-tests` directory, and is itself just a specific case of #45444, which affected Rally, Kibana CI, and presumably any other test suite that sent a non-trivial amount of data between nodes.  You could raise an issue like this one for every test in `native-multi-node-tests`, for example [`ModelPlotIT`](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(branch,test,build_url,build-id),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:kuery,query:'class%20:%20%22org.elasticsearch.xpack.ml.integration.ModelPlotsIT%22'),sort:!(time,desc))), [`OverallBucketsIT`](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(branch,test,build_url,build-id),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:kuery,query:'class%20:%20%22org.elasticsearch.xpack.ml.integration.OverallBucketsIT%22'),sort:!(time,desc))) and [`RestoreModelSnapshotIT`](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(branch,test,build_url,build-id),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:kuery,query:'class%20:%20%22org.elasticsearch.xpack.ml.integration.RestoreModelSnapshotIT%22'),sort:!(time,desc))) also failed regularly between 9th and 12th August.  This is why I suggested muting the entire directory in https://github.com/elastic/elasticsearch/issues/45405#issuecomment-520378806.

It's true that there are sporadic failures for other reasons, for example the one that shows up in the chart on 14th August (after #45444 was fixed).  But unlike the problem that affected the whole of `native-multi-node-tests` these more intermittent failures will differ between the precise tests that are failing, so a catch-all issue at the suite level isn't really a good way to track them.  The people doing test triage on the days when these sporadic failures occur should be opening specific issues for the specific failures.

**Comment by davidkyle** [comment]
Looking at the recent failures I'm happy they were all due to #45444 and fixed on the 12th. I checked the  outlier on the 14th and it was a PR build based on an older commit.

Looking back at the last 30 days of failures, there were a number around 30th July, I putting those down to branching issues as that was the time of the 7.2.1 & 7.3.0 release. There were a bunch of failures on 16th July, they were all PR builds which failed with hundreds of test failures. 

Closing this issue. The evidence doesn't point to any flakiness in the test all the failures are due to external issues.
